templates/index.html

```diff
@@ -3,337 +3,467 @@
-                        commit.remove();
-                        humanTime.remove();
-                    } else {
-                        imageTag.innerHTML = `<code>${domainJson['imageTag']}</code>`;
-                        commit.innerHTML = `<a href="https://github.com/${domainRepositories[domainJson['domain']]}/commit/${domainJson['commitId']}">${domainJson['commitId']}</a>`;
-                        humanTime.textContent = domainJson['humanTime'];
-                        if (release) {
-                            release.innerHTML = `<a style="width: 100%" class="p-button--neutral" href="https://jenkins.canonical.com/webteam/securityRealm/commenceLogin?from=%2Fwebteam%2Fjob%2Fdeploy-to-${domainJson['domain']}%2Fparambuild%2F%3Fimage_tag%3D${domainJson['imageTag']}">` +
-                                                `  Release` +
-                                                `</a>`;
-                        }
-                    }
-                }
-            }
+      if ('error' in domainJson) {
+        imageTag.setAttribute('colspan', '3');
+        imageTag.innerHTML = `<small style="color: red; font-style: italic">${domainJson.error}</small>`;
+        commit.remove();
+        humanTime.remove();
+      } else {
+        imageTag.innerHTML = `<code>${domainJson['imageTag']}</code>`;
+        commit.innerHTML = `<a href="https://github.com/${domainRepositories[domainJson['domain']]}/commit/${domainJson['commitId']}">${domainJson['commitId']}</a>`;
+        humanTime.textContent = domainJson['humanTime'];
+        humanTime.dataset.sort = domainJson['epochTime'];
+        if (release) {
+          release.innerHTML = `<a style="width: 100%" class="p-button--neutral" href="https://jenkins.canonical.com/webteam/securityRealm/commenceLogin?from=%2Fwebteam%2Fjob%2Fdeploy-to-${domainJson['domain']}%2Fparambuild%2F%3Fimage_tag%3D${domainJson['imageTag']}">` +
+            `  Release` +
+            `</a>`;
+          }
+        }
+      }
+    }
 
-            document.querySelectorAll('[data-domain]').forEach(
-                function (domainRow) {
-                    let domain = domainRow.dataset.domain;
-                    fetch(`/domain-info.json?domain=${domain}`)
-                        .then(async function (response) {
-                            return response.json();
-                        })
-                        .then(writeDomainDetails(domainRow))
-                }
-            )
-        </script>
+    document.querySelectorAll('[data-domain]').forEach(
+    function (domainRow) {
+      let domain = domainRow.dataset.domain;
+      fetch(`/domain-info.json?domain=${domain}`)
+      .then(async function (response) {
+        return response.json();
+      })
+      .then(writeDomainDetails(domainRow))
+    }
+    )
+  </script>
 
-        <div class="p-strip is-shallow is-bordered">
-          <div class="row">
-            <h3>Staging domains not yet supported</h3>
+  <script>
+    (function () {
```

Comment:
Wouldn't it be nice if these were modules?

Reply:
This would be nice but doesn't get use much as we have no build pipeline or tests.